plugins/exporter.koplugin/main.lua

```diff
@@ -226,6 +226,7 @@ function Exporter:exportClippings(clippings)
         -- full file path without extension
         clipping_filepath = clipping_dir .. "/" ..
             util.getSafeFilename(self.ui.bookinfo:expandString(clipping_filename, file, timestamp), nil, nil, -1)
+        clipping_filepath = clipping_filepath:gsub("[\n\r]", ", "):gsub("\t", " ")
```

Comment:
I meant to handle it in getSafeFilename so you can pass a string with a newline in it. I could see a similar problem inadvertently coming from OPDS or NewsDownloader or something, or perhaps user input. Of course special behavior like replacing newline with `, ` would then have to be done first.

Incidentally, wouldn't a Windows `\r\n` become `, , ` as written?

Reply:
> special behavior like replacing newline with `, `

What is the common behaviour? Replace with the white space?

Reply:
Okay, to summarize:
-in export, \n and \r\n are replaced with ;
-in util, \n and \r\n and \t are replaced with space
Right?